File: Aplikacja_Mobilna/App_IK/app/src/main/java/inteligenty_zamek/app_ik/GenerationCertyficatActivity.java
package inteligenty_zamek.app_ik;

import android.content.Intent;
import android.content.res.TypedArray;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.ViewFlipper;

import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.NameValuePair;
import org.apache.http.client.ClientProtocolException;
import org.apache.http.client.HttpClient;
import org.apache.http.client.entity.UrlEncodedFormEntity;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.impl.client.DefaultHttpClient;
import org.apache.http.message.BasicNameValuePair;
import org.apache.http.util.EntityUtils;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class GenerationCertyficatActivity extends BaseActivity implements AdapterView.OnItemSelectedListener {

    private String[] navMenuTitles;
    private TypedArray navMenuIcons;

    String[] items = { "Certyfikat dla użytkownika", "Certyfikat dla gościa"};
    ViewFlipper viewFlipper;
    Spinner spin,spin2;
    TextView name;
    TextView surname;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_generation_certyficat);
        navMenuTitles= getResources().getStringArray(R.array.nav_drawer_items);
        navMenuIcons =getResources().obtainTypedArray(R.array.nav_drawer_icons);
        set(navMenuTitles,navMenuIcons);

        viewFlipper = (ViewFlipper) findViewById(R.id.myViewFlipper);
        name = (TextView) this.findViewById(R.id.warning_icologin);
        surname = (TextView) this.findViewById(R.id.warning_icologin);
        viewFlipper.setDisplayedChild(0);
        spin = (Spinner) findViewById(R.id.spinnerChangeAdminGenerationCertyficat);
        spin2 = (Spinner) findViewById(R.id.spinnerChangeAdminGenerationCertyficat2);
        spin.setOnItemSelectedListener(this);
        spin2.setOnItemSelectedListener(this);

        ArrayAdapter aa = new ArrayAdapter(
                this,
                android.R.layout.simple_spinner_item,
                items);

        ArrayAdapter aa2 = new ArrayAdapter(
                this,
                android.R.layout.simple_spinner_item,
                items);

        aa.setDropDownViewResource(
                android.R.layout.simple_spinner_dropdown_item);
        spin.setAdapter(aa);

        aa2.setDropDownViewResource(
                android.R.layout.simple_spinner_dropdown_item);
        spin2.setAdapter(aa2);
        User user=((GlobalClassContainer) getApplication()).getUser();
        new HTTPRequestDownloadKey(user).execute();
        new HTTPRequestDownloadUser(user).execute();
        final Button register = (Button) findViewById(R.id.buttonRangeGenerateCertyficat);
        register.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent intent = new Intent(GenerationCertyficatActivity.this,GenerateCertyficatRangeActivity.class);
                startActivity(intent);
            }
        });


        final Button generate = (Button) findViewById(R.id.generujAdmin);
        generate.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                EditText from=(EditText) findViewById(R.id.inputfromAdmingenerate);
                EditText to=(EditText) findViewById(R.id.inputtoadminGenerateKey);
                final Spinner spiner1 = (Spinner) findViewById(R.id.spinnerLoks);
                final Spinner spiner2 = (Spinner) findViewById(R.id.spinnerUser);
                EditText name=(EditText) findViewById(R.id.inputName);
                String fromstring=from.getText().toString();
                String tostring=to.getText().toString();
                int lokposition=spiner1.getSelectedItemPosition();
                int userposition=spiner2.getSelectedItemPosition();
                String namestring=name.getText().toString();
                new HTTPRequest(((GlobalClassContainer) getApplication()).getUser() ,fromstring ,tostring,lokposition,userposition,namestring ).execute();


            }
        });


    }



    public void onItemSelected(AdapterView<?> parent, View v, int position,
                               long id) {
        if(position==1){spin2.setSelection(1);} else {spin.setSelection(0);}
        viewFlipper.setDisplayedChild(position);

    }

    public void onNothingSelected(AdapterView<?> parent) {




    }



    public class HTTPRequestDownloadKey extends AsyncTask<Void, Void, String> {
        User user;

        public HTTPRequestDownloadKey(User x){
            user=x;
        }
        @Override
        protected String doInBackground(Void... params) {
            HttpClient httpclient = new DefaultHttpClient();

            String adres="http://"+ ((GlobalClassContainer) getApplication()).getSerwerIP()+":8080/api/download/all_locks/";

            HttpPost httppost = new HttpPost(adres);

            try {
                // Add your data
                List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(2);
                nameValuePairs.add(new BasicNameValuePair("login", user.getLogin()));

                nameValuePairs.add(new BasicNameValuePair("token",  ((GlobalClassContainer) getApplication()).getSession()));


                httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));

                // Execute HTTP Post Request
                HttpResponse response = httpclient.execute(httppost);
                HttpEntity entity = response.getEntity();
                String responseString = EntityUtils.toString(entity, "UTF-8");

                return responseString;
            } catch (ClientProtocolException e) {
                // TODO Auto-generated catch block
            } catch (IOException e) {
                // TODO Auto-generated catch block
            }

            return null;
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            JSONObject jObj = null;
            try {
                jObj = new JSONObject(response);

                JSONArray arrJson = jObj.getJSONArray("data");
                ((GlobalClassContainer) getApplication()).getUser().addLockList(arrJson);

                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Spinner spiner1 = (Spinner) findViewById(R.id.spinnerLoks);
                        final ArrayList<String> listkey = new ArrayList<String>();
                        final SpinnerAdapter spinerListKeyAdapter=new SpinnerAdapter(GenerationCertyficatActivity.this, R.layout.spinner_row,  listkey,  listkey);
                        spiner1.setAdapter(spinerListKeyAdapter);
                        Lock[] list=((GlobalClassContainer) getApplication()).getUser().getLockslist();
                        for(int i=0; i<list.length; i++)
                        {
                            spinerListKeyAdapter.add(list[i].getName());
                            spinerListKeyAdapter.notifyDataSetChanged();
                        }
                    }
                });

            } catch (JSONException e) {

            }
        }
    }



    public class HTTPRequest extends AsyncTask<Void, Void, String> {
        User user;
        String from; String to; int idloks; int iduser; String nameCert;
        public HTTPRequest(User x,String from, String to, int idloks, int iduser, String nameCert ){
            user=x;
            this.from=from;
            this.to=to;
            this.idloks=idloks;
            this.iduser=iduser;
            this.nameCert=nameCert;
        }
        @Override
        protected String doInBackground(Void... params) {
            HttpClient httpclient = new DefaultHttpClient();


            String adres="http://"+ ((GlobalClassContainer) getApplication()).getSerwerIP()+":8080/api/admin/generate_new_certificate/";

            HttpPost httppost = new HttpPost(adres);

            try {
                // Add your data
              final List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(2);
                nameValuePairs.add(new BasicNameValuePair("login", user.getLogin()));
                nameValuePairs.add(new BasicNameValuePair("token",  ((GlobalClassContainer) getApplication()).getSession()));
                nameValuePairs.add(new BasicNameValuePair("user_id",  ((GlobalClassContainer) getApplication()).getUserlist()[iduser].getIdUser()));
                nameValuePairs.add(new BasicNameValuePair("lock_id",  ((GlobalClassContainer) getApplication()).getUser().getLockslist()[idloks].getIdKey()));
                GenerationCertyficatActivity.this.runOnUiThread(new Runnable() {
                    public void run() {
                        nameValuePairs.add(new BasicNameValuePair("name", name.getText().toString() ));
                        nameValuePairs.add(new BasicNameValuePair("surname", surname.getText().toString() ));
                    }
                });


                nameValuePairs.add(new BasicNameValuePair("from_date",from ));
                nameValuePairs.add(new BasicNameValuePair("to_date",to ));


                if( ((GlobalClassContainer) getApplication()).getMondayList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getMondayList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("monday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("monday", " "));}

                if( ((GlobalClassContainer) getApplication()).getTuesdayList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getTuesdayList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("tuesday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("tuesday", " "));}


                if( ((GlobalClassContainer) getApplication()).getWednesdayList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getWednesdayList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("wednesday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("wednesday", " "));}


                if( ((GlobalClassContainer) getApplication()).getThurstdayList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getThurstdayList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("thursday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("thursday", " "));}


                if( ((GlobalClassContainer) getApplication()).getFridyList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getFridyList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("friday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("friday", " "));}

                if( ((GlobalClassContainer) getApplication()).getSaturdayList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getSaturdayList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("saturday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("saturday", " "));}

                if( ((GlobalClassContainer) getApplication()).getSundayList()!=null) {
                    ArrayList<String> list=((GlobalClassContainer) getApplication()).getSundayList();
                    String result="";
                    for (String s : list)
                    {
                        result += s + ";";
                    }
                    nameValuePairs.add(new BasicNameValuePair("sunday", result));
                }
                else {nameValuePairs.add(new BasicNameValuePair("sunday", " "));}

                nameValuePairs.add(new BasicNameValuePair("is_pernament","0" ));




                httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
                //Log.i("tutaj"," 5");

                HttpResponse response = httpclient.execute(httppost);
                HttpEntity entity = response.getEntity();
                String responseString = EntityUtils.toString(entity, "UTF-8");

                return responseString;
            } catch (ClientProtocolException e) {
                // TODO Auto-generated catch block
            } catch (IOException e) {
                // TODO Auto-generated catch block
            }

            return null;
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            JSONObject jObj = null;
            try {
                jObj = new JSONObject(response);

            } catch (JSONException e) {

            }


        }
    }





    public class HTTPRequestDownloadUser extends AsyncTask<Void, Void, String> {
        User user;

        public HTTPRequestDownloadUser(User x){
            user=x;
        }
        @Override
        protected String doInBackground(Void... params) {
            HttpClient httpclient = new DefaultHttpClient();

            String adres="http://"+ ((GlobalClassContainer) getApplication()).getSerwerIP()+":8080/api/download/all_user/";

            HttpPost httppost = new HttpPost(adres);

            try {
                // Add your data
                List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(2);
                nameValuePairs.add(new BasicNameValuePair("login", user.getLogin()));

                nameValuePairs.add(new BasicNameValuePair("token",  ((GlobalClassContainer) getApplication()).getSession()));


                httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));

                // Execute HTTP Post Request
                HttpResponse response = httpclient.execute(httppost);
                HttpEntity entity = response.getEntity();
                String responseString = EntityUtils.toString(entity, "UTF-8");

                return responseString;
            } catch (ClientProtocolException e) {
                // TODO Auto-generated catch block
            } catch (IOException e) {
                // TODO Auto-generated catch block
            }

            return null;
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            JSONObject jObj = null;
            try {
                jObj = new JSONObject(response);

                JSONArray arrJson = jObj.getJSONArray("data");
                ((GlobalClassContainer) getApplication()).addUserList(arrJson);

                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Spinner spiner1 = (Spinner) findViewById(R.id.spinnerUser);
                        final ArrayList<String> listkey = new ArrayList<String>();
                        final SpinnerAdapter spinerListKeyAdapter=new SpinnerAdapter(GenerationCertyficatActivity.this, R.layout.spinner_row,  listkey,  listkey);
                        spiner1.setAdapter(spinerListKeyAdapter);
                        User[] list=((GlobalClassContainer) getApplication()).getUserlist();
                        for(int i=0; i<list.length; i++)
                        {
                            spinerListKeyAdapter.add(list[i].getLogin());
                            spinerListKeyAdapter.notifyDataSetChanged();
                        }
                    }
                });

            } catch (JSONException e) {

            }
        }
    }






}

File: Aplikacja_Mobilna/App_IK/app/src/main/java/inteligenty_zamek/app_ik/sampledata/GenerationCertyficatActivity.kt
package inteligenty_zamek.app_ik.sampledata

import android.content.Intent
import android.content.res.TypedArray
import android.os.AsyncTask
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import android.widget.ViewFlipper
import inteligenty_zamek.app_ik.GenerateCertyficatRangeActivity

import org.apache.http.NameValuePair
import org.apache.http.client.ClientProtocolException
import org.apache.http.client.entity.UrlEncodedFormEntity
import org.apache.http.client.methods.HttpPost
import org.apache.http.impl.client.DefaultHttpClient
import org.apache.http.message.BasicNameValuePair
import org.apache.http.util.EntityUtils
import org.json.JSONException
import org.json.JSONObject

import java.io.IOException
import java.util.ArrayList

import inteligenty_zamek.app_ik.rest_class.GlobalClassContainer
import inteligenty_zamek.app_ik.rest_class.User
import inteligenty_zamek.app_ik.Navigation.BaseActivity
import inteligenty_zamek.app_ik.R
import inteligenty_zamek.app_ik.SpinnerAdapter

class GenerationCertyficatActivity : BaseActivity(), AdapterView.OnItemSelectedListener {

    private var navMenuTitles: Array<String>? = null
    private var navMenuIcons: TypedArray? = null

    internal var items = arrayOf("Certyfikat dla użytkownika", "Certyfikat dla gościa")
    internal var viewFlipper: ViewFlipper?=null
    internal var spin: Spinner?=null
    internal var spin2: Spinner?=null
    internal var name2: TextView?=null
    internal var surname: TextView?=null
    internal var nam: String?=null
    internal var surnam: String?=null
    internal var pozlock = -1
    internal var pozus = -1
    internal var lock = ""
    internal var login = ""
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_generation_certyficat)
        navMenuTitles = resources.getStringArray(R.array.nav_drawer_items)
        navMenuIcons = resources.obtainTypedArray(R.array.nav_drawer_icons)
        set(navMenuTitles, navMenuIcons)
/*
        viewFlipper = findViewById(R.id.myViewFlipper) as ViewFlipper
        name2 = this.findViewById(R.id.editText7) as TextView
        surname = this.findViewById(R.id.editText8) as TextView
        viewFlipper.displayedChild = 0
        spin = findViewById(R.id.spinnerChangeAdminGenerationCertyficat) as Spinner
        spin2 = findViewById(R.id.spinnerChangeAdminGenerationCertyficat2) as Spinner
        spin.onItemSelectedListener = this
        spin2.onItemSelectedListener = this

        try {
            lock = intent.getStringExtra("lock")
            login = intent.getStringExtra("name")
        } catch (e: Exception) {
        }

        val aa = ArrayAdapter(
                this,
                android.R.layout.simple_spinner_item,
                items)

        val aa2 = ArrayAdapter(
                this,
                android.R.layout.simple_spinner_item,
                items)

        aa.setDropDownViewResource(
                android.R.layout.simple_spinner_dropdown_item)
        spin.adapter = aa

        aa2.setDropDownViewResource(
                android.R.layout.simple_spinner_dropdown_item)
        spin2.adapter = aa2
        val user = (application as GlobalClassContainer).user
        HTTPRequestDownloadKey(user).execute()
        HTTPRequestDownloadUser(user).execute()
        val register = findViewById(R.id.buttonRangeGenerateCertyficat) as Button
        register.setOnClickListener {
            val intent = Intent(this@GenerationCertyficatActivity, GenerateCertyficatRangeActivity::class.java)
            startActivity(intent)
        }


        val generate = findViewById(R.id.generujAdmin) as Button
        generate.setOnClickListener {
            val from = findViewById(R.id.inputfromAdmingenerate) as EditText
            val to = findViewById(R.id.inputtoadminGenerateKey) as EditText
            val spiner1 = findViewById(R.id.spinnerLoks) as Spinner
            val spiner2 = findViewById(R.id.spinnerUser) as Spinner
            val name = findViewById(R.id.inputName) as EditText
            val fromstring = from.text.toString()
            val tostring = to.text.toString()
            val lokposition = spiner1.selectedItemPosition
            val userposition = spiner2.selectedItemPosition
            val namestring = name.text.toString()
            nam = name2.text.toString()
            surnam = surname.text.toString()
            HTTPRequest((application as GlobalClassContainer).user, fromstring, tostring, lokposition, userposition, namestring).execute()
        }
*/

    }


    override fun onItemSelected(parent: AdapterView<*>, v: View, position: Int,
                                id: Long) {
        if (position == 1) {
          // spin2.setSelection(1)
        } else {
        //    spin.setSelection(0)
        }
       // viewFlipper.displayedChild = position

    }

    override fun onNothingSelected(parent: AdapterView<*>) {


    }


    inner class HTTPRequestDownloadKey(internal var user: User) : AsyncTask<Void, Void, String>() {
        override fun doInBackground(vararg params: Void): String? {
            val httpclient = DefaultHttpClient()

            val adres = "http://" + (application as GlobalClassContainer).serwerIP + ":8080/api/download/all_locks/"

            val httppost = HttpPost(adres)

            try {
                // Add your data
                val nameValuePairs = ArrayList<NameValuePair>(2)
                nameValuePairs.add(BasicNameValuePair("login", user.login))

                nameValuePairs.add(BasicNameValuePair("token", (application as GlobalClassContainer).session))


                httppost.entity = UrlEncodedFormEntity(nameValuePairs)

                // Execute HTTP Post Request
                val response = httpclient.execute(httppost)
                val entity = response.entity

                return EntityUtils.toString(entity, "UTF-8")
            } catch (e: ClientProtocolException) {
                // TODO Auto-generated catch block
            } catch (e: IOException) {
                // TODO Auto-generated catch block
            }

            return null
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        override fun onPostExecute(response: String) {
            super.onPostExecute(response)
            var jObj: JSONObject? = null
            try {
                jObj = JSONObject(response)

                val arrJson = jObj.getJSONArray("data")
                (application as GlobalClassContainer).user.addLockList(arrJson)

                runOnUiThread {
                    val spiner1 = findViewById(R.id.spinnerLoks) as Spinner
                    val listkey = ArrayList<String>()
                    val spinerListKeyAdapter = SpinnerAdapter(this@GenerationCertyficatActivity, R.layout.spinner_row, listkey, listkey)
                    spiner1.adapter = spinerListKeyAdapter
                    val usrLog = (application as GlobalClassContainer).user.login
                    val list = (application as GlobalClassContainer).user.getLockslist(usrLog)
                    for (i in list.indices) {
                        try {
                            if (lock == list[i].name.toString()) {
                                spiner1.setSelection(i)
                            }
                        } catch (e: Exception) {
                        }

                        spinerListKeyAdapter.add(list[i].name)
                        spinerListKeyAdapter.notifyDataSetChanged()
                    }
                }

            } catch (e: JSONException) {

            }

        }
    }


    inner class HTTPRequest(internal var user: User, internal var from: String, internal var to: String, internal var idloks: Int, internal var iduser: Int, internal var nameCert: String) : AsyncTask<Void, Void, String>() {
        override fun doInBackground(vararg params: Void): String? {
            val httpclient = DefaultHttpClient()


            val adres = "http://" + (application as GlobalClassContainer).serwerIP + ":8080/api/admin/generate_new_certificate/"

            val httppost = HttpPost(adres)

            try {
                // Add your data
                val nameValuePairs = ArrayList<NameValuePair>(2)
                nameValuePairs.add(BasicNameValuePair("login", user.login))
                nameValuePairs.add(BasicNameValuePair("token", (application as GlobalClassContainer).session))
                nameValuePairs.add(BasicNameValuePair("user_id", (application as GlobalClassContainer).userlist[iduser].getIdUser()))
                val login = (application as GlobalClassContainer).user.login
                nameValuePairs.add(BasicNameValuePair("lock_id", (application as GlobalClassContainer).user.getLockslist(login)[idloks].idKey))

                nameValuePairs.add(BasicNameValuePair("name", nam))
                nameValuePairs.add(BasicNameValuePair("surname", surnam))



                nameValuePairs.add(BasicNameValuePair("from_date", from))
                nameValuePairs.add(BasicNameValuePair("to_date", to))


                if ((application as GlobalClassContainer).mondayList != null) {
                    val list = (application as GlobalClassContainer).mondayList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("monday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("monday", " "))
                }

                if ((application as GlobalClassContainer).tuesdayList != null) {
                    val list = (application as GlobalClassContainer).tuesdayList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("tuesday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("tuesday", " "))
                }


                if ((application as GlobalClassContainer).wednesdayList != null) {
                    val list = (application as GlobalClassContainer).wednesdayList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("wednesday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("wednesday", " "))
                }


                if ((application as GlobalClassContainer).thurstdayList != null) {
                    val list = (application as GlobalClassContainer).thurstdayList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("thursday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("thursday", " "))
                }


                if ((application as GlobalClassContainer).fridyList != null) {
                    val list = (application as GlobalClassContainer).fridyList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("friday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("friday", " "))
                }

                if ((application as GlobalClassContainer).saturdayList != null) {
                    val list = (application as GlobalClassContainer).saturdayList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("saturday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("saturday", " "))
                }

                if ((application as GlobalClassContainer).sundayList != null) {
                    val list = (application as GlobalClassContainer).sundayList
                    var result = ""
                    for (s in list) {
                        result += s + ";"
                    }
                    nameValuePairs.add(BasicNameValuePair("sunday", result))
                } else {
                    nameValuePairs.add(BasicNameValuePair("sunday", " "))
                }

                nameValuePairs.add(BasicNameValuePair("is_pernament", "0"))




                httppost.entity = UrlEncodedFormEntity(nameValuePairs)
                //Log.i("tutaj"," 5");

                val response = httpclient.execute(httppost)
                val entity = response.entity

                return EntityUtils.toString(entity, "UTF-8")
            } catch (e: ClientProtocolException) {
                // TODO Auto-generated catch block
            } catch (e: IOException) {
                // TODO Auto-generated catch block
            }

            return null
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        override fun onPostExecute(response: String) {
            super.onPostExecute(response)
            var jObj: JSONObject? = null
            try {
                jObj = JSONObject(response)

            } catch (e: JSONException) {

            }


        }
    }


    inner class HTTPRequestDownloadUser(internal var user: User) : AsyncTask<Void, Void, String>() {
        override fun doInBackground(vararg params: Void): String? {
            val httpclient = DefaultHttpClient()

            val adres = "http://" + (application as GlobalClassContainer).serwerIP + ":8080/api/download/all_user/"

            val httppost = HttpPost(adres)

            try {
                // Add your data
                val nameValuePairs = ArrayList<NameValuePair>(2)
                nameValuePairs.add(BasicNameValuePair("login", user.login))

                nameValuePairs.add(BasicNameValuePair("token", (application as GlobalClassContainer).session))


                httppost.entity = UrlEncodedFormEntity(nameValuePairs)

                // Execute HTTP Post Request
                val response = httpclient.execute(httppost)
                val entity = response.entity

                return EntityUtils.toString(entity, "UTF-8")
            } catch (e: ClientProtocolException) {
                // TODO Auto-generated catch block
            } catch (e: IOException) {
                // TODO Auto-generated catch block
            }

            return null
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        override fun onPostExecute(response: String) {
            super.onPostExecute(response)
            var jObj: JSONObject? = null
            try {
                jObj = JSONObject(response)

                val arrJson = jObj.getJSONArray("data")
                (application as GlobalClassContainer).addUserList(arrJson)

                runOnUiThread {
                    val spiner1 = findViewById(R.id.spinnerUser) as Spinner
                    val listkey = ArrayList<String>()
                    val spinerListKeyAdapter = SpinnerAdapter(this@GenerationCertyficatActivity, R.layout.spinner_row, listkey, listkey)
                    spiner1.adapter = spinerListKeyAdapter
                    val list = (application as GlobalClassContainer).userlist
                    for (i in list.indices) {
                        try {
                            if (login == list[i].name.toString()) {
                                spiner1.setSelection(i)
                            }
                        } catch (e: Exception) {
                        }

                        spinerListKeyAdapter.add(list[i].login)
                        spinerListKeyAdapter.notifyDataSetChanged()
                    }
                }

            } catch (e: JSONException) {

            }

        }
    }


}